From: Don Dutile <ddutile@redhat.com>
Date: Thu, 17 Jul 2008 19:19:50 -0400
Subject: [xen] PV: Makefile and Kconfig additions
Message-id: 487FD396.30505@redhat.com
O-Subject: [RHEL5.3 PATCH 4/5]: Makefile and Kconfig additions to build XEN PV-on-HVM drivers in bare metal / FV guest kernel
Bugzilla: 442991
RH-Acked-by: Chris Lalancette <clalance@redhat.com>
RH-Acked-by: Bill Burns <bburns@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

BZ 442991 -- Include xenpv-driver in bare metal kernel rpm.

The following set of files add the pv-on-hvm drivers into
the default builds of the i686 & x86_64 bare metal (FV) kernels.

I guess they don't have to be a separate patch, but
that's how I did it for RHEL4.7.  Not much to test,
other than build capability.

Please review and ACK.

- Don

diff --git a/arch/i386/Kconfig b/arch/i386/Kconfig
index d04fcc6..8d050cf 100644
--- a/arch/i386/Kconfig
+++ b/arch/i386/Kconfig
@@ -1222,6 +1222,10 @@ source "crypto/Kconfig"
 
 source "drivers/xen/Kconfig"
 
+if !X86_XEN
+source "drivers/xenpv_hvm/Kconfig"
+endif
+
 source "lib/Kconfig"
 
 #
diff --git a/arch/x86_64/Kconfig b/arch/x86_64/Kconfig
index 651d6cb..fe3be72 100644
--- a/arch/x86_64/Kconfig
+++ b/arch/x86_64/Kconfig
@@ -721,4 +721,8 @@ source "crypto/Kconfig"
 
 source "drivers/xen/Kconfig"
 
+if !X86_64_XEN
+source "drivers/xenpv_hvm/Kconfig"
+endif
+
 source "lib/Kconfig"
diff --git a/drivers/Makefile b/drivers/Makefile
index 2857358..083ac5b 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -32,6 +32,7 @@ obj-$(CONFIG_NUBUS)		+= nubus/
 obj-$(CONFIG_ATM)		+= atm/
 obj-$(CONFIG_PPC_PMAC)		+= macintosh/
 obj-$(CONFIG_XEN)		+= xen/
+obj-$(CONFIG_XEN_PV_ON_HVM)	+= xenpv_hvm/
 obj-$(CONFIG_IDE)		+= ide/
 obj-$(CONFIG_FC4)		+= fc4/
 obj-$(CONFIG_SCSI)		+= scsi/
